src/validation.cpp

```diff
@@ -1070,6 +1070,11 @@ static bool AcceptToMemoryPoolWorker(
                     auto it = mempoolReferral.Find(referral->alias);
                     while (it.first != it.second) {
                         const auto duplicate_referral = it.first->GetSharedEntryValue();
+                        it.first++;
+
+                        if (referral->GetHash() == duplicate_referral->GetHash()) {
```

Comment:
quick question, is it possible to generate a referral that may have a different hash but same address returned by GetAddress?

Reply:
sure, just change a field other then address or type. 
we do not check if address already beaconed in `AcceptToMemoryPool`, but we do it in `ConnectBlock`
looks there could be a problem if you want to beacon same address with different alias or parent address, as we send invites by address or alias